selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/PushingChangesTest.java

```diff
@@ -81,138 +81,92 @@
   @Inject private TestGitHubKeyUploader testGitHubKeyUploader;
   @Inject private TestUserPreferencesServiceClient testUserPreferencesServiceClient;
   @Inject private TestGitHubServiceClient gitHubClientService;
+  @Inject private TestGitHubRepository testRepo;
+  @Inject private TestProjectServiceClient testProjectServiceClient;
+  @Inject private TestCommandServiceClient testCommandServiceClient;
+  @Inject private CommandsPalette commandsPalette;
 
   @BeforeClass
-  public void prepare() throws Exception {
+  public void setUp() throws Exception {
     testGitHubKeyUploader.updateGithubKey();
     testUserPreferencesServiceClient.addGitCommitter(gitHubUsername, productUser.getEmail());
+
+    Path entryPath = Paths.get(getClass().getResource("/projects/git-pull-test").getPath());
+    testRepo.addContent(entryPath);
+    prepareAmendCommitCommandByRestApi();
     ide.open(ws);
+    projectExplorer.waitProjectExplorer();
+    git.importJavaApp(testRepo.getSshUrl(), PROJECT_NAME, BLANK);
+    projectExplorer.waitItem(PROJECT_NAME);
+  }
+
+  @AfterClass
+  public void deleteRepo() throws IOException {
+    testRepo.delete();
```

Comment:
It is a excess action, because deleting the test repo already has in the 'TestGitHubRepository' with a '@PreDestroy' annotation